client/app.jsx

```diff
@@ -1,8 +1,43 @@
 import React from 'react';
 import Home from './pages/home';
+import { parseRoute } from './lib';
+import Header from './components/header';
+import AppContext from './lib/app-context';
 
 export default class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash),
+      user: null
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', () => {
+      this.setState({ route: parseRoute(window.location.hash) });
+    });
+  }
+
+  renderPage(props) {
```

Comment:
When you call `renderPage` you aren't passing it anything so it doesn't need a parameter here. So just `renderPage() {` is good.